src/Symfony/Component/Security/Core/Authorization/Voter/RoleVoter.php

```diff
@@ -37,6 +37,13 @@ public function __construct($prefix = 'ROLE_')
      */
     public function supportsAttribute($attribute)
     {
+        if (is_array($attribute)) {
+            return false;
+        }
+        if (is_object($attribute) && !method_exists($attribute, '__toString')) {
+            return false;
+        }
+
         return 0 === strpos($attribute, $this->prefix);
```

Comment:
> It shouldn't crash for non string values. It should abstain

Sorry, you're right. However i think for 2.7 `supportAttribute` should also consider `RoleInterface` due mixed attribute support.